doozer/doozerlib/cli/scan_osh.py

```diff
@@ -0,0 +1,157 @@
+
+            else:
+                if "el7" in nvr:
+                    rhel_version = 7
+                elif "el8" in nvr:
+                    rhel_version = 8
+                elif "el9" in nvr or nvr.startswith("rhcos"):
+                    rhel_version = 9
+                else:
+                    self.runtime.logger.error("Invalid RHEL version")
+                    return
+
+                cmd = cmd_template.format(config=f"rhel-{rhel_version}-x86_64", nvr=nvr)
+
+            message = f"Ran command: {cmd}"
+
+            if not self.dry_run:
+                fire_and_forget(self.runtime.cwd, cmd)
+            else:
+                message = "[DRY RUN] " + message
+
+            self.runtime.logger.info(message)
+
+    async def run(self):
+        if not self.specific_nvrs:
+            tags = self.runtime.get_errata_config()["brew_tag_product_version_mapping"].keys()
+            for tag in tags:
+                major, minor = self.runtime.group.replace("openshift-", "").split(".")
+                self.brew_tags.append(tag.format(MAJOR=major, MINOR=minor))
+
+            builds = []
+            if self.last_brew_event:
+                # If the --since field is specified, find all the builds that are after the specified brew event
+                for tag in self.brew_tags:
+                    builds += self.get_tagged_all(tag=tag)
+
+                # Sort the builds based on the event ID by descending order so that latest is always on top
+                builds = sorted(builds, key=lambda x: x["create_event"], reverse=True)
+                builds = [build for build in builds if build["create_event"] > self.last_brew_event]
+
+            else:
+                # If no --since field is specified, find all the builds that have been tagged into our candidate tags
+                for tag in self.brew_tags:
+                    builds += self.get_tagged_latest(tag=tag)
+                builds = sorted(builds, key=lambda x: x["create_event"], reverse=True)
+
+            nvrs = [build["nvr"] for build in builds]
+            nvr_brew_mapping = [(build["nvr"], build["create_event"]) for build in builds]
+
+            if nvr_brew_mapping:
+                self.runtime.logger.info(f"NVRs to trigger scans for {nvr_brew_mapping}")
+                pass
+
+            if builds:
+                latest_event_id = nvr_brew_mapping[0][1]
+
+                self.nvrs = nvrs
+
+                # Return back the latest brew event ID
+                cprint(latest_event_id)
+            else:
+                self.runtime.logger.warning(
+                    f"No new NVRs have been found since last brew event: {self.last_brew_event}")
+                return None
+        else:
+            self.runtime.logger.info(f"Triggering scans for this particular list of NVRs: {self.specific_nvrs}")
+
+            self.nvrs = self.specific_nvrs
+
+        if self.check_triggered:
+            await self.untriggered_nvrs()
```

Comment:
It's not apparent what is happening here. Something like 
`nvrs = self.get_untriggered_nvrs(nvrs)` would make it clear

Reply:
https://github.com/openshift-eng/art-tools/pull/51/commits/7c715b307aa02efa5d521ca0e931c9a2ae18766e